### ISSUE TITLE
solve throws on small SMatrix with Measurements.jl entries

### ISSUE THREAD

**Issue by mapclyps** [JuliaArrays/StaticArrays.jl#939]
I was using using backslash on some SMatrices with Measurements.jl entries and when I used smaller matrices with sizes 2x2 and 3x3 it started to throw an error.

Measurements v2.6.0
StaticArrays v1.2.7

MWE
```julia
julia> using Measurements

julia> using StaticArrays

julia> n=5;a=SMatrix{n,n}(rand(n,n).±0.1);

julia> a\a; #works

julia> n=3;a=SMatrix{n,n}(rand(n,n).±0.1);

julia> a\a
ERROR: setindex!() with non-isbitstype eltype is not supported by StaticArrays. Consider using SizedArray.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] setindex!
    @ ~/.julia/packages/StaticArrays/AHT47/src/MArray.jl:99 [inlined]
  [3] macro expansion
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:66 [inlined]
  [4] _setindex!_scalar
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:46 [inlined]
  [5] setindex!
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:42 [inlined]
  [6] macro expansion
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:357 [inlined]
  [7] _setindex!
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:320 [inlined]
  [8] setindex!
    @ ~/.julia/packages/StaticArrays/AHT47/src/indexing.jl:274 [inlined]
  [9] _solve
    @ ~/.julia/packages/StaticArrays/AHT47/src/solve.jl:36 [inlined]
 [10] \(a::SMatrix{3, 3, Measurement{Float64}, 9}, b::SMatrix{3, 3, Measurement{Float64}, 9})
    @ StaticArrays ~/.julia/packages/StaticArrays/AHT47/src/solve.jl:1
 [11] top-level scope
    @ REPL[6]:1
    ```

**Comment by mateuszbaran** [comment]
It looks like `MArray` doesn't work well with non-isbits types, and actually the implementation of `\` for 3x3 matrices makes a temporary `MArray`. I think `similar` should return `SizedArray` instead for non-isbits `SArray`.

**Comment by mapclyps** [comment]
This was also discussed in #799 and then later fixed in #1196. It works now
